firebase-installations/api.txt

```diff
@@ -20,110 +20,10 @@ package com.google.firebase.installations {
-    method @Nullable public abstract String getRefreshToken();
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallation.RegistrationStatus getRegistrationStatus();
-    method public abstract long getTokenCreationEpochInSecs();
-    method public boolean isErrored();
-    method public boolean isNotGenerated();
-    method public boolean isRegistered();
-    method public boolean isUnregistered();
-    method public boolean shouldAttemptMigration();
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallationEntry.Builder toBuilder();
-    method @NonNull public com.google.firebase.installations.local.PersistedInstallationEntry withAuthToken(@NonNull String, long, long);
-    method @NonNull public com.google.firebase.installations.local.PersistedInstallationEntry withClearedAuthToken();
-    method @NonNull public com.google.firebase.installations.local.PersistedInstallationEntry withFisError(@NonNull String);
-    method @NonNull public com.google.firebase.installations.local.PersistedInstallationEntry withNoGeneratedFid();
-    method @NonNull public com.google.firebase.installations.local.PersistedInstallationEntry withRegisteredFid(@NonNull String, @NonNull String, long, @Nullable String, long);
-    method @NonNull public com.google.firebase.installations.local.PersistedInstallationEntry withUnregisteredFid(@NonNull String);
-    field @NonNull public static com.google.firebase.installations.local.PersistedInstallationEntry INSTANCE;
-  }
-
-  public abstract static class PersistedInstallationEntry.Builder {
-    ctor public PersistedInstallationEntry.Builder();
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallationEntry build();
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallationEntry.Builder setAuthToken(@Nullable String);
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallationEntry.Builder setExpiresInSecs(long);
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallationEntry.Builder setFirebaseInstallationId(@NonNull String);
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallationEntry.Builder setFisError(@Nullable String);
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallationEntry.Builder setRefreshToken(@Nullable String);
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallationEntry.Builder setRegistrationStatus(@NonNull com.google.firebase.installations.local.PersistedInstallation.RegistrationStatus);
-    method @NonNull public abstract com.google.firebase.installations.local.PersistedInstallationEntry.Builder setTokenCreationEpochInSecs(long);
-  }
-
 }
 
 package com.google.firebase.installations.remote {
```

Comment:
Hide this package as well?

afaict from your api review, the only approved api is the following:

```
public abstract class InstallationTokenResult {
    ctor public InstallationTokenResult();

    // not part of api review, remove pls
    // method @NonNull public static com.google.firebase.installations.InstallationTokenResult.Builder builder();
    method @NonNull public abstract String getToken();
    method @NonNull public abstract long getTokenCreationTimestamp();
    method @NonNull public abstract long getTokenExpirationTimestamp();

    // not part of api review, remove pls
    // method @NonNull public abstract com.google.firebase.installations.InstallationTokenResult.Builder toBuilder();
  }

public class FirebaseInstallations implements com.google.firebase.installations.FirebaseInstallationsApi {
    method @NonNull public com.google.android.gms.tasks.Task<java.lang.Void> delete();
    method @NonNull public com.google.android.gms.tasks.Task<java.lang.String> getId();
    method @NonNull public static com.google.firebase.installations.FirebaseInstallations getInstance();
    method @NonNull public static com.google.firebase.installations.FirebaseInstallations getInstance(@NonNull com.google.firebase.FirebaseApp);

    // I think this should take a boolean as an argument according to your api review. pls confirm
    method @NonNull public com.google.android.gms.tasks.Task<com.google.firebase.installations.InstallationTokenResult> getToken(@com.google.firebase.installations.FirebaseInstallationsApi.AuthTokenOption int);
  }
```

Everything else should be hidden afaict.

Reply:
Updated the file.